File: 1700/solution.py
class Solution:
    def countStudents(self, students: list[int], sandwiches: list[int]) -> int:
        ones = 0
        zeroes = 0
        for student in students:
            if student == 0:
                zeroes += 1
            else:
                ones += 1
        for i, sandwich in enumerate(sandwiches):
            if sandwich == 0 and zeroes:
                zeroes -= 1
            elif sandwich == 1 and ones:
                ones -= 1
            else:
                return len(sandwiches) - i
        return 0

File: 1700/solution.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int zeroes = 0;
        int ones = 0;
        int S = sandwiches.length;
        for (int student : students) {
            if (student == 1)
                ones++;
            else
                zeroes++;
        }
        for (int i = 0; i < S; i++) {
            if (sandwiches[i] == 0 && zeroes > 0)
                zeroes--;
            else if (sandwiches[i] == 1 && ones > 0)
                ones--;
            else
                return S - i;
        }
        return 0;
    }
}

// time O(st + sa)
// space O(1)